pkg/scheduler/scheduler.go

```diff
@@ -227,10 +227,18 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 
 		if mode == flavorassigner.Preempt && len(e.preemptionTargets) == 0 {
 			log.V(2).Info("Workload requires preemption, but there are no candidate workloads allowed for preemption", "preemption", cq.Preemption)
-			// we use resourcesToReserve to block capacity up to either the nominal capacity,
-			// or the borrowing limit when borrowing, so that a lower priority workload cannot
-			// admit before us.
-			cq.AddUsage(resourcesToReserve(e, cq))
+			// we reserve capacity if we are not sure that
+			// we can reclaim the capacity later via
+			// preemption. Otherwise, we allow workloads
+			// to schedule, confident we can preempt them
+			// later.
+			if !preemption.CanAlwaysPreempt(cq) {
```

Comment:
What if fairSharing case? IIUC, preemptee can preempt workloads only when the usage situation fit in the preemptionStrategies.

Reply:
Since the policy is set to Reclaim=Any, CQ will be able to preempt if scores indicate the assignments are unfair. I believe this is WAI

Reply:
Regardless of drs, the flavor assigner always say `Preempt` mode in this case: https://github.com/kubernetes-sigs/kueue/blob/36b0094a94dfd847b48e9eca50a9d1d5896ac4af/pkg/scheduler/flavorassigner/flavorassigner_test.go#L1828

So, could you check if the unexpected reclaiming never happened when the usage situation fit in the fhairSharing preemptionStrategies?